scripts/add_repo.sh

```diff
@@ -50,10 +51,27 @@ if [ ! -d "$remote_dir" ]; then
 
 	# add the subtree
 	git remote update $remote_name
-	ref=$(git show-ref remotes/$remote_name/master -s)
-	git subtree add --prefix="$remote_dir" "$remote_name" --squash master
-	sh -c "cd $remote_dir && go mod edit -replace $downstream_repo=../../"
-	git add $remote_dir/go.mod
+	ref=$(git show-ref remotes/$remote_name/${target_branch} -s)
+	git subtree add --prefix="$remote_dir" "$remote_name" --squash ${target_branch}
+	new_mod=$(cd $remote_dir && go list -m)
+
+	for staged_dep in $(ls $staging_dir | grep -v "^$remote_name$"); do
+		staged_mod=$(cd $staging_dir/$staged_dep && go list -m)
+		grep "$staged_mod" $remote_dir/go.mod && sh -c "cd $remote_dir && \
+								go mod edit -require $staged_mod@v0.0.0-00010101000000-000000000000 && \
+								go mod edit -replace $staged_mod=../$staged_dep"
+		grep "$new_mod" $staging_dir/$staged_dep/go.mod && sh -c "cd $staging_dir/$staged_dep && \
+								go mod edit -require $new_mod@v0.0.0-00010101000000-000000000000 && \
+								go mod edit -replace $new_mod=../$remote_name && \
+								go mod vendor && \
+								git add go.mod go.sum vendor"
+	done
+
+	sh -c "cd $remote_dir && \
+		go mod edit -replace $downstream_repo=../.. && \
```

Comment:
Minor nit: I believe we likely want to avoid hardcoding that `../..` directory path. We may want to create something like a `ROOT_DIR` that points to the root directory path by grabbing the current working (e.g. `scripts`) directory and appending a `../` to that.

Reply:
That's a good point, it would help with directory structure changes